inbm/dispatcher-agent/tests/unit/schedule/test_sqlite_manager.py

```diff
@@ -0,0 +1,100 @@
+        with self.assertRaisesRegex(DispatcherException, 
+                                    'Error adding single schedule to database: NOT NULL constraint failed: single_schedule.start_time'):
+            self.db.create_schedule(ss1)
+    
+    def test_raise_exception_when_create_single_schedule_with_no_manifests(self):
+        ss1 = SingleSchedule(request_id="REQ123", 
+                            start_time="2024-01-01T00:00:00",
+                            end_time="2024-01-02T00:00:00", 
+                            manifests=[])
+        with self.assertRaisesRegex(DispatcherException, 
+                                    'Error: At least one manifest is required for the schedule.  Manifests list is empty.'):
+            self.db.create_schedule(ss1)
+                    
+    def test_raise_exception_when_create_repeated_schedule_with_no_manifests(self):
+        rs1 = RepeatedSchedule(request_id="REQ123",
+                            cron_duration=None,
+                            cron_minutes="*/3",
+                            manifests=[])
+        with self.assertRaisesRegex(DispatcherException, 
+                                    'Error: At least one manifest is required for the schedule.  Manifests list is empty.'):
+            self.db.create_schedule(rs1)
+        
+    def test_create_simple_schedule(self):
+        ss1 = SingleSchedule(request_id="REQ123", 
+                            start_time="2024-01-01T00:00:00", 
+                            end_time="2024-01-02T00:00:00", 
+                            manifests=["MANIFEST1", "MANIFEST2"])
+        ss2 = SingleSchedule(request_id="REQ234", 
+                            start_time="2024-05-01T00:00:00", 
+                            manifests=["MANIFEST3", "MANIFEST4"])
+
+        self.db.create_schedule(ss1)
+        self.db.create_schedule(ss2)
+        res1 = self.db.select_single_schedule_by_request_id("REQ123")
+        res2 = self.db.select_single_schedule_by_request_id("REQ234")
+        
+        self.assertEqual(res1[0].id, 1)
+        self.assertEqual(res1[0].request_id, "REQ123")
+        self.assertEqual(res1[0].start_time, "2024-01-01T00:00:00")
+        self.assertEqual(res1[0].end_time, "2024-01-02T00:00:00")
+        self.assertEqual(res1[0].manifests, ["MANIFEST1", "MANIFEST2"])
+        
+        self.assertEqual(res2[0].request_id, "REQ234")
+        self.assertEqual(res2[0].start_time, "2024-05-01T00:00:00")
+        #self.assertEqual(res2[0].end_time, None)
+        self.assertEqual(res2[0].manifests, ["MANIFEST3", "MANIFEST4"])
+        
+    def test_create_repeated_schedule(self):
+        rs1 = RepeatedSchedule(request_id="REQ123", 
+                            cron_duration="*",
+                            cron_minutes="0",
+                            cron_hours="0",
+                            cron_day_week="1-5",
+                            manifests=["MANIFEST1", "MANIFEST2"])
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Add validation in SqliteManager to check for null or invalid cron fields in RepeatedSchedule before attempting to create the schedule in the database.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        # Inside SqliteManager.create_schedule method
        if isinstance(schedule, RepeatedSchedule):
            if not self._validate_cron_fields(schedule):
                raise DispatcherException('Invalid cron fields in RepeatedSchedule')
```